File: interview/prob150.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        
        def eval(a, b, op):
            if op == "+":
                return a + b
            elif op == "-":
                return a - b
            elif op == "*":
                return a * b
            else:
                return int(a / b)
            
        stack = []
        
        for t in tokens:
            if t in "+-*/":
                b, a = stack.pop(), stack.pop()
                stack.append(eval(a, b, t))
            else:
                stack.append(int(t))
                
        return stack[0]

File: interview/prob150.go
import (
    "strings"
    "strconv"
)

func evalRPN(tokens []string) int {
    
    stack := []int{}
    
    for _, t := range tokens {
        if strings.Contains("+-*/", t) {
            sz := len(stack)
            a, b := stack[sz-2], stack[sz-1]
            c := eval(a, b, t)
            stack = append(stack[:sz-2], c)
        } else {
            v, _ := strconv.Atoi(t)
            stack = append(stack, v)
        }
    }
    
    return stack[0]
    
}

func eval(a int, b int, op string) int {
    if op == "+" {
        return a + b
    } else if op == "-" {
        return a - b
    } else if op == "*" {
        return a * b
    } else {
        return a / b
    }
}
